### PR TITLE
Masterbar: Require and use 'jetpack-masterbar' package in jetpack-mu-wpcom

### LINKED ISSUE #36918: Masterbar package: Require the package in both the Jetpack plugin and the mu-wpcom plugin
```[tasklist]
### Tasks
- [x] Masterbar package: Require the package in the Jetpack plugin
- [ ] Masterbar package: Require the package in the jetpack-mu-wpcom package
```

### PR DESCRIPTION
**Pull request by fgiannar**
~~**Important**: This PR should be merged after JP `13.6-a.3` or **later** is released on AT. Context: https://github.com/Automattic/jetpack/pull/37812#issuecomment-2172322573~~

Jetpack `13.6-a.5` is now released to AT: p9o2xV-4mW-p2

Masterbar: Require and use 'jetpack-masterbar' package in jetpack-mu-wpcom

This is needed as we are slowly deprecating the feature for self-hosted sites.

## Proposed changes:

* Masterbar package: Refactored `Admin_Color_Schemes` to incorporate the logic from the corresponding feature in `jetpack-mu-wpcom`
* Masterbar package: Update `Main` class so that it behaves consistently with the WPCOM masterbar mu-plugin. Aka with Simple sites in mind.
* Jetpack-mu-wpcom package: Load Masterbar features on WoA sites (this is only temp as the we plan to make it available to Simple sites too)
* Jetpack-mu-wpcom package: Refactor `admin-color-schemes.php` to simply instantiate `Admin_Color_Schemes` upon loading **only for classic Simple sites** (also temp).

### Other information:

- [ ] Have you written new tests for your changes, if applicable?
- [ ] Have you checked the E2E test CI results, and verified that your changes do not break them?
- [ ] Have you tested your changes on WordPress.com, if applicable (if so, you'll see a generated comment below with a script to run)?

## Jetpack product discussion

pfwV0U-3U-p2

## Does this pull request change what data or activity we track or use?

No

## Testing instructions:
**We'll need to repeat the testing from https://github.com/Automattic/jetpack/pull/37676 with the following extras:**

### Simple Sites

We are mostly interested in classic Simple sites. These are the ones where `wpcom_is_nav_redesign_enabled()` will be `true`.  This means that `return get_blog_option( YOUR_WPCOM_BLOG_ID, 'wpcom_admin_interface' )` returns `wp-admin`. I was under the impression that switching to the Classic view would reflect this but it wasn't the case for me, so I had to manually set the `wpcom_admin_interface` option to `wp-admin` for my Simple site.

**User Profile**
- Verify that the `Admin Color Scheme` setting in `profile.php` screen looks the same with/without the PR applied. 
- Try switching Color Schemes and verify the changes are applied (no need to save). 

### Atomic Sites

**We'll need to test with the provided instructions below but also with Jetpack on the current JP version on WoA aka `13.6-a.5` and the WordPress.com Features plugin on this branch via Jetpack beta. This will simulate the period between the next wpcomsh release and the next Jetpack AT release.**

We'll need to test both classic and default WoA sites. You can toggle this option via Settings->General in `wp-admin` for a **classic** WoA site

Screenshot 2024-06-13 at 14 29 19

 You can toggle this option via Settings->General in Calypso for a **Default** WoA site:

Screenshot 2024-06-18 at 13 06 21

~~TODO: On a WoA default site, the Appearance->Additonal CSS link redirects you to wp-admin with an error message that you don't have access. Is this intended?~~

### CODE REVIEW

**Review by coder-karen — COMMENTED** [review]
This looks great!

On reviewing the code, it all looks good.

I tested on Simple, Atomic, self-hosted. I think most of what I found may not even be issues, but I'll add them here:

> Important: Without the PR, only jetpack-core-color-schemes-overrides-sidebar-notice-css is loaded but my understanding based on the linked Jetpack product discussion above is that this is the desired behaviour.

On my test Atomic, neither was loaded when the PR wasn't applied. The color schemes looked and behaved the same though.

> For block based themes, "Customize" should not be present under "Appearance".
(from the linked PR)

Testing on a Simple site, using block based theme (Twenty Twenty Four), I see Customize under Appearance. However I do with trunk too. Same with Classic or Default interface style. Atomic behaves as the PR description suggests though.

> On a WoA default site, the Appearance->Additonal CSS link redirects you to wp-admin with an error message that you don't have access. Is this intended?

I couldn't replicate that. I know I mentioned in Slack I'd seem the message briefly, but that was while testing the Custom CSS PR, not this one.

**Comment by fgiannar** [comment]
Thanks for the review, @coder-karen !

Unfortunately I'm able to reproduce a Fatal when testing the `admin-menu` endpoint on a WoA site on JP `13.6-a.1` and the WordPress.com Features plugin on this branch via Jetpack beta.

`Cannot redeclare Automattic\Jetpack\Masterbar\should_customize_nav()`

I think the safest approach here would be extracting the suggested change in the endpoint from this PR to another PR, then merge it so it's available on the next AT release and merging this one afterwards so it becomes available to WoA sites via the corresponding `wpcomsh` update.

**Comment by fgiannar** [comment]
Many many thanks, @coder-karen !

> Important: Without the PR, only jetpack-core-color-schemes-overrides-sidebar-notice-css is loaded but my understanding based on the linked Jetpack product discussion above is that this is the desired behaviour.

This applies to **classic Simple sites** and the `profile.php` screen in `wp-admin`.

> Testing on a Simple site, using block based theme (Twenty Twenty Four), I see Customize under Appearance. However I do with trunk too. Same with Classic or Default interface style. Atomic behaves as the PR description suggests though.

It would be nice to get a confirmation around the intended behaviour from the product teams cc @mmtr 

> I couldn't replicate that. I know I mentioned in Slack I'd seem the message briefly, but that was while testing the Custom CSS PR, not this one.

I'm also not able to reproduce this one anymore but will keep an eye in case it happens again while testing.

**Review by mmtr — DISMISSED** [review]
Impressive work, @fgiannar.

This works as expected to me, with no regressions in the behavior when I compare it against `trunk`.

Tested in 5 different environments:
- WP.com Simple Default: Color schemes are correctly overridden and I can only change them in Calypso
- WP.com Simple Classic: Color schemes are correctly overridden and I can only change them in WP Admin
- WP.com Atomic Default: Color schemes are correctly overridden and I can only change them in Calypso
- WP.com Atomic Classic: Color schemes are correctly overridden and I can only change them in WP Admin
- Self-hosted sites: Color schemes are correctly overridden when the toolbar module is enabled and I can change them in WP Admin. The selected color scheme is not synced with Calypso, as expected.

EDIT: Somehow I thought I only had to test the admin color schemes changes. I'll review the rest of features tomorrow.

**Review by mmtr — APPROVED** [review]
Made sure to test everything now, and it works as expected 🚀 

> Testing on a Simple site, using block based theme (Twenty Twenty Four), I see Customize under Appearance. However I do with trunk too. Same with Classic or Default interface style. Atomic behaves as the PR description suggests though.

Yeah, this is unrelated to these changes, because it happens in `trunk` as well. Due to a bug, the "Appearance > Customize" menu is visible on Simple Default sites when visiting a WP Admin page. It does not happen in other scenarios (Simple Default visiting Calypso, Simple Classic visiting any page).

**Review by coder-karen — APPROVED** [review]
Retested color schemes in particular on all site types, default and classic styles, as that was the only change since my last review. All tests well :)

File: projects/packages/masterbar/src/class-main.php [hunk 1]
```diff
@@ -27,21 +27,25 @@ public static function init() {
...
 
 		new Admin_Color_Schemes();
 
-		if ( ( new Host() )->is_woa_site() ) {
+		if ( $host->is_wpcom_platform() ) {
 			new Inline_Help();
 			require_once __DIR__ . '/wp-posts-list/bootstrap.php';
-			require_once __DIR__ . '/profile-edit/bootstrap.php';
 			require_once __DIR__ . '/nudges/bootstrap.php';
 		}
```
Comment (mmtr) [inline projects/packages/masterbar/src/class-main.php:43]: Something I just noted, should we keep the WoA check here? I see Simple sites are already loading these features in their specific mu-plugin:

fbhepr%2Skers%2Sjcpbz%2Sjc%2Qpbagrag%2Szh%2Qcyhtvaf%2Sznfgreone.cuc%3Se%3Qns1125q9%23272%2Q291-og

They tested well for me though, so it seems that loading twice these features don't affect the end result, but maybe we should play safe?

Comment (mmtr) [reply projects/packages/masterbar/src/class-main.php:43]: Oh never mind, I realized we still don't load this package on Simple sites. We're just making the package ready to make the switch from the mu-plugin to the package.

**Pull request merged**